jtransc-gen-d/src/com/jtransc/gen/d/DTarget.kt

```diff
@@ -355,4 +355,17 @@ class DGenerator(injector: Injector) : CommonGenerator(injector) {
 	override fun genExprCastChecked(e: String, from: AstType.Reference, to: AstType.Reference): String {
 		return "checkCast!(${to.targetNameRef})($e)"
 	}
+
+	override fun genBody2WithFeatures(method: AstMethod, body: AstBody): Indenter = Indenter {
+		if(method.modifiers.isSynchronized) {
+			line("try{")
```

Comment:
Though many targets have `try{ ... } finally { ... }` syntax, not all targets support it. So I am afraid I cannot put it into CommonGenerator.